recipe/build-mpi.sh

```diff
@@ -33,15 +38,26 @@ if [[ "$target_platform" == osx-* ]]; then
     wrapper_ldflags='-Wl,-rpath,${libdir}'
 fi
 
-# UCX support
+# UCX and UCC support
 build_with_ucx=""
-if [[ "$target_platform" == linux-* ]]; then
+build_with_ucc=""
+if [[ "$target_platform" == linux-* && "$target_platform" != linux-ppc64le ]]; then
+    echo "Build with UCX/UCC support"
     build_with_ucx="--with-ucx=$PREFIX"
+    build_with_ucc="--with-ucc=$PREFIX"
+fi
+
+# PMIx support
+build_with_pmix=""
+if [[ "$target_platform" == linux-* && "$target_platform" != linux-ppc64le* ]]; then
+    echo "Build with PMIx support"
+    build_with_pmix="--with-pmix=internal"
 fi
```

Comment:
PMIx was added to support the use of OpenMPI on high-performance computing environments with Slurm, ensuring efficient process management and resource allocation
I tested it and it works well

Reply:
I meant why we didn't add it earlier :) 

But, see also https://github.com/conda-forge/openmpi-feedstock/pull/175#issuecomment-2374252917, I don't think this works yet

Reply:
@j34ni The ask was to investigate why you added `pmix` in the host section but we still see it shown as "internal" instead of "yes" (which would mean the conda-forge `pmix` is detected and to be used at build time), not to change to `--with-pmix=internal` here. 

As the name implied, Open MPI already has a copy of PMIx and so when the build system cannot detect any installed in the user environment, it'd use the private copy. See e.g. the build log from the main branch: https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=991597&view=logs&j=d0d954b5-f111-5dc4-4d76-03b6c9d0cf7e&t=779a0c34-7b19-5ccb-5da7-fe4579c91392&l=5814

Reply:
This is because I originally wanted to use an external installation of PMIx (to offer more flexibility, as for libevent and hwloc) and hence to configure with `--with-pmix=$PREFIX`, however the https://github.com/openpmix/pmix-feedstock is not ready yet so I temporarily used `--with-pmix=internal` as a placeholder (rather than not even mentioning it at all and letting people doing an `ompi_info` think that it was not supported)

Reply:
Oh I did not realize there's no `pmix` on conda-forge. Note that we cannot build a package that would depend on packages from another channel.

> so I temporarily used `--with-pmix=internal` as a placeholder

In-line comments would not help issue tracking (and there's no comment added here to describe that this is for placeholding). I suggest opening an issue to track the need, and undo what's done here:
```suggestion
```

> letting people doing an `ompi_info` think that it was not supported

As described above, it is supported. Lack of `--with-pmix=internal` in `Configure command line` does not mean PMIx was not used at build time. See the linked build log from the main branch.

Reply:
> I suggest opening an issue to track the need

xref: #178